archunit/src/test/java/com/tngtech/archunit/core/domain/DependencyTest.java

```diff
@@ -71,6 +71,31 @@ public void Dependency_from_throws_declaration() {
                 .contains("Method <" + origin.getFullName() + "> throws type <" + IOException.class.getName() + ">");
     }
 
+    @Test
+    public void Dependency_from_instanceof_check_in_method() {
```

Comment:
I am now missing a test for an `instanceof` check in a constructor, since you have meanwhile removed `InstanceofCheckTest` (which I think is good, because as you noticed there are other tests that cover the same aspects). Or have I missed it somehow?
Also you could consider making this a `@DataProvider` test, where one `JavaConstructor`, one `JavaMethod` and one `JavaStaticInitializer` are passed in from which you expect the same behavior (i.e. each of them doing the same `instanceof` check so the assertion can be uniform). (might need to parameterize the line number as well though :wink:)

Reply:
Done. The `@DataProvider` is really nice 😃 
I had to extract the line number per regex from the `SourceCodeLocation` as there is no other way to get it from a `JavaCodeUnit`.